code/modules/mob/living/silicon/robot/robot_items.dm

```diff
@@ -95,4 +95,59 @@
 	desc = "By retracting limbs and tucking in its head, a combat android can roll at high speeds."
 	icon = 'icons/obj/decals.dmi'
 	icon_state = "shock"
-	powerneeded = 25
\ No newline at end of file
+	powerneeded = 25
+
+
+//APC power adapter for cyborgs, adapted from apc_powercord code
+//Allows cyborgs to recharge via APC, at first faster than a stock cyborg recharging station, upgraded stations are faster.
+obj/item/ccharger
+	name = "APC power adapter"
+	desc = "A cyborg specific power adapter, allowing cyborgs to siphon power from APCs to recharge their powercells"
+	icon = 'icons/obj/cyborg.dmi'
+	icon_state = "c-charger"
+	flags = NOBLUDGEON
+
+/obj/item/ccharger/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	if(!istype(target, /obj/machinery/power/apc) || !issilicon(user) || !proximity_flag)
+		return ..()
+	user.changeNext_move(CLICK_CD_MELEE)
+	var/obj/machinery/power/apc/A = target
+	var/mob/living/silicon/robot/H = user
+	if(H.cell)
+		if(A.emagged || A.stat & BROKEN)
+			do_sparks(3, 1, A)
+			to_chat(H, "<span class='warning'>The APC power currents surge erratically, damaging your chassis!</span>")
+			H.adjustFireLoss(10,0)
+		else if(A.cell && A.cell.charge > 0)
+			if(H.cell.charge >= H.cell.maxcharge)
+				to_chat(user, "<span class='warning'>You are already fully charged!</span>")
+			else
+				INVOKE_ASYNC(src, .proc/powerdraw_loop, A, H)
+		else
+			to_chat(user, "<span class='warning'>There is no charge to draw from that APC.</span>")
+	else
+		to_chat(user, "<span class='warning'>You lack a cell in which to store charge!</span>")
+
+/obj/item/ccharger/proc/powerdraw_loop(obj/machinery/power/apc/A, mob/living/silicon/robot/H)
+	H.visible_message("<span class='notice'>[H] inserts a power connector into \the [A].</span>", "<span class='notice'>You begin to draw power from \the [A].</span>")
+	while(do_after(H, 10, target = A))
+		if(loc != H)
+			to_chat(H, "<span class='warning'>You must keep your connector out while charging!</span>")
+			break
+		if(A.cell.charge == 0)
+			to_chat(H, "<span class='warning'>\The [A] has no more charge.</span>")
+			break
+		A.charging = TRUE
+		if(A.cell.charge >= 200)
+			H.cell.give(200 * H.cell.rating) // replenishment scales with higher capacity cells
+			A.cell.use(200) // as to not rapidly drain and depower rooms 
```

Comment:
This would be better implemented as a subtype of `apc_powercord`, where you replace the actions in this part of the loop with a subtype-specific action - for the human version give "nutrition", but for the borg version provide cell charge

Reply:
I would have to refactor apc_powercord just to accommodate the new item, which beyond the scope of this pr.

Reply:
"Refactor" isn't a scary word. All you need to do is add a couple of more procs to make the behavior match between what you copy pasted it from.
https://gist.github.com/Crazylemon64/2f16dd661944326562e5f4d5059d67cc

Granted, now that you've changed your implementation from a copypasta to a proper port, this isn't directly applicable, but keep in mind: When you do a copypasta, this kind of "refactor" is expected. It is definitely NOT out of scope.

Reply:
The reason why we want these refactors on copypasta'd code is to avoid having multiple implementations for the same behavior. Say that we want, as a hypothetical example, the charger to hack the user if used on an emagged APC. We would need to implement it on each copy of the cable, meaning more places to check, and more chances to make mistakes. A similar argument exists for bugfixing - if you have many copies, you might only fix a bug on a single instance when it exists on them all.

Reply:
sorta shows my relative inexperience and I'll keep that in mind for the future.